apps/src/gamelab/GameLabSprite.js

```diff
@@ -42,210 +41,220 @@ module.exports.createSprite = function (x, y, width, height) {
-   */
-  s._syncAnimationSizes = function (animations, currentAnimation) {};
-
-  Object.defineProperty(s, 'frameDelay', {
-    enumerable: true,
-    get: function () {
-      if (s.animation) {
-        return s.animation.frameDelay;
-      }
-    },
-    set: function (value) {
-      if (s.animation) {
-        s.animation.frameDelay = value;
-      }
-    }
-  });
-
-  // Overriding these allows users to set a width for
+  // Overriding these allows users to set a width and height for
   // an animated sprite the same way they would an unanimated sprite.
   Object.defineProperty(s, 'width', {
     enumerable: true,
     configurable: true,
-    get: function () {
-      if (s._internalWidth === undefined) {
-        return 100;
-      } else if (s.animation) {
-        return s._internalWidth * s._horizontalStretch;
-      } else {
-        return s._internalWidth;
-      }
-    },
-    set: function (value) {
-      if (s.animation) {
-        s._horizontalStretch = value / s._internalWidth;
-      } else {
-        s._internalWidth = value;
-      }
-    }
+    get: getWidth,
+    set: setWidth
   });
-
-  // Overriding these allows users to set a height for
-  // an animated sprite the same way they would an unanimated sprite.
   Object.defineProperty(s, 'height', {
     enumerable: true,
     configurable: true,
-    get: function () {
-      if (s._internalHeight === undefined) {
-        return 100;
-      } else if (s.animation) {
-        return s._internalHeight * s._verticalStretch;
-      } else {
-        return s._internalHeight;
-      }
-    },
-    set: function (value) {
-      if (s.animation) {
-        s._verticalStretch = value / s._internalHeight;
-      } else {
-        s._internalHeight =  value;
-      }
-    }
+    get: getHeight,
+    set: setHeight
   });
 
-  // p5.play stores width unscaled, but users in
-  // Game Lab should have access to a scaled version.
-  s.getScaledWidth = function () {
-    return s.width * s.scale;
-  };
-
-  // p5.play stores height unscaled, but users in
-  // Game Lab should have access to a scaled version.
-  s.getScaledHeight = function () {
-    return s.height * s.scale;
-  };
+  // Attach our custom/override methods to the sprite
+  s.setAnimation = setAnimation.bind(s, this);
+  s.frameDidChange = frameDidChange;
+  s.pointTo = pointTo;
+  s._getScaleX = _getScaleX;
+  s._getScaleY = _getScaleY;
+  s._syncAnimationSizes = _syncAnimationSizes;
+  s.getScaledWidth = getScaledWidth;
+  s.getScaledHeight = getScaledHeight;
+  s._collideWith = _collideWith.bind(s, this);
+  s._collideWithOne = _collideWithOne.bind(s, this);
+  s.createGroupState = createGroupState;
+  s.bounceOff = bounceOff;
+  s.isTouching = isTouching;
+  // Overriding overlap, collide, displace, bounce, to work with our group states.
+  s.overlap = overlap;
+  s.collide = collide;
+  s.displace = displace;
+  s.bounce = bounce;
+  s.play = play;
```

Comment:
Another nice side effect of this structure is that it's really obvious how little ordering matters for all these assignments.